File: javascript/claude/p122.js
// Allocate two chunks of memory (arrays of size 10)
let chunk1 = new Array(10);
let chunk2 = new Array(10);

// Fill first chunk with some data
for (let i = 0; i < 10; i++) {
    chunk1[i] = i;
}

// Copy contents of first chunk into second chunk
chunk2 = [...chunk1];

console.log("chunk1:", chunk1);
console.log("chunk2:", chunk2);

// Free (dereference) the first chunk
chunk1 = null;

// Free (dereference) the second chunk
chunk2 = null;

File: java/claude/p122.java
public class MemoryAllocation {
    public static void main(String[] args) {
        // Allocate two chunks of memory (arrays of size 10)
        int[] chunk1 = new int[10];
        int[] chunk2 = new int[10];
        
        // Fill first chunk with some data
        for (int i = 0; i < 10; i++) {
            chunk1[i] = i;
        }
        
        // Copy contents of first chunk into second chunk
        System.arraycopy(chunk1, 0, chunk2, 0, 10);
        
        System.out.println("chunk1: " + java.util.Arrays.toString(chunk1));
        System.out.println("chunk2: " + java.util.Arrays.toString(chunk2));
        
        // Set references to null for garbage collection
        chunk1 = null;
        chunk2 = null;
    }
}
